pennylane/gradients/pulse_generator_gradient.py

```diff
@@ -0,0 +1,236 @@
+
+    # Compute the Jacobian of _compute_matrix, giving the Jacobian of the real and imag parts
+    # The output is a tuple, with one entry per parameter, each of which has the axes
+    # (mat_dim, mat_dim, *parameter_shape)
+    jac_real, jac_imag = jax.jacobian(_compute_matrix_split, argnums=0)(op.data)
+
+    # Compute the matrix of the pulse itself and conjugate it. Skip the transposition of the adjoint
+    # The output has the shape (mat_dim, mat_dim)
+    U_dagger = qml.math.conj(_compute_matrix([qml.math.detach(d) for d in op.data]))
+
+    # Compute U^\dagger @ \partial U / \partial \theta_k
+    # For each entry ``j`` in the tuple ``jac``,
+    # 1. Contract U_dagger with j along the axes [0], [0]. This effectively transposes
+    #    U_dagger, which we skipped above.
+    # 2. After contraction, the axes are (mat_dim, mat_dim, *parameter_shape).
+    # 3. Move the first two axis to the last two positions.
+    moveax = partial(qml.math.moveaxis, source=(0, 1), destination=(-2, -1))
+    return tuple(
+        moveax(qml.math.tensordot(U_dagger, j_r + 1j * j_i, axes=[[0], [0]]))
+        for j_r, j_i in zip(jac_real, jac_imag)
+    )
+
+
+def _one_parameter_paulirot_coeffs(generators, num_wires):
+    r"""Compute the Pauli coefficients of the one-parameter group generators that reproduce
+    the partial derivatives of a parametrized evolution. The coefficients correspond
+    to the decomposition into (rescaled) Pauli word generators as used by ``PauliRot``
+    gates.
+    In particular, compute :math:`U` and :math:`\partial U / \partial \theta_k`
+    and recombine them into :math:`\Omega_k = U^\dagger \partial U / \partial \theta_k`
+
+    Args:
+        generators (tuple[tensor_like]): Generators of the one-parameter groups that
+            reproduce the partial derivatives of the parametrized evolution of interest.
+        num_wires (int): Number of wires the parametrized evolution acts on.
+
+    Returns:
+        tuple[tensor_like]: Coefficients of the provided generators in the Pauli basis,
+        modified by a factor of ``2j`` (see warning below).
+
+    .. warning::
+
+        This function includes a prefactor ``2j`` in its output compared to the "standard" Pauli
+        basis coefficients. That is, for the effective generator :math:`\frac{1}{3}iX`, which has
+        the Pauli basis coefficient :math:`\frac{1}{3}i`, this function will compute the
+        number :math:`-\frac{2}{3}`. This is in order to accomodate the use of ``qml.PauliRot``
+        in the pulse generator differentiation pipeline.
+
+    .. note::
+
+        Currently, this function work via tensor multiplication, costing
+        :math:`\mathcal{O}(n16^N)` scalar multiplications, where :math:`N` is the
+        number of qubits the evolution acts on and :math:`n` is the number of (scalar) parameters
+        of the evolution. This can be improved to :math:`\mathcal{O}(nN4^N}` by using the
+        fast Walsh-Hadamard transform.
```

Comment:
I'm not sure how helpful this note is, and this docstring is very long

Reply:
Okay, I'll skip it. There is a similar comment in `_pauli_decompose` docstring anyways :)